backend/Cache.js

```diff
@@ -0,0 +1,61 @@
+class Cache {
+  constructor(ttl = 3600, maxSize = 100) {
+    this.cache = new Map();
+    this.ttl = ttl * 1000; // Convert TTL to milliseconds.
+    this.maxSize = maxSize; // Maximum number of items in cache
+
+    // Periodically clean up expired entries
+    setInterval(this.cleanup.bind(this), this.ttl);
+  }
+
+  get(key) {
+    const cachedItem = this.cache.get(key);
+    if (!cachedItem) {
+      return null;
+    }
+
+    const [value, expiryTime] = cachedItem;
+
+    if (Date.now() > expiryTime) {
+      // Item has expired
+      this.cache.delete(key);
+      return null;
+    }
+
+    // Update the item's position to implement LRU
+    this.cache.delete(key);
+    this.cache.set(key, [value, expiryTime]);
+
+    return value;
+  }
+
+  set(key, value) {
+    if (this.cache.size >= this.maxSize) {
+      // Evict the least recently used item
+      const firstKey = this.cache.keys().next().value;
+      this.cache.delete(firstKey);
```

Comment:
`firstKey` returns the value, not the key, so this won't delete the key from the cache? You'd need to pass the key as param itself.

Reply:
I'm not aiming to delete the key, but to replace the value once the max size is hit. Is this method logical or am i still doing it wrong? Please let me know.

Reply:
I see. I think that makes sense. You will want to delete the key and add a new key the end of the map to preserve correct insertion order. 

Adding a new key with a new value after deleting the oldest one will put the new key at the end of the list, marking it as "most recently used".

Reply:
@mylesdomingo 

Just checked and this works fine, the value in this instance refers to the value of the key itself. This is because the `.next()` function returns the next key in this format: `{value: 'key', done: false}`